iidm/iidm-converter-api/src/main/java/com/powsybl/iidm/tools/AbstractImportingTool.java

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.tools;
+
+import com.powsybl.iidm.import_.ImportConfig;
+import com.powsybl.tools.Tool;
+import org.apache.commons.cli.CommandLine;
+
+/**
+ * @author Miora Ralambotiana <miora.ralambotiana at rte-france.com>
+ */
+public abstract class AbstractImportingTool implements Tool {
```

Comment:
I happen to come across this PR:
I think having an abstract tool is not the best design, because you will not be able to add other standard options this way (multiple inheritance).

I would propose somethink like "Option objects" which would be able to do 2 things:
1. Add option definitions
2. Read an objet from the command line (generic parameter)

This way, tools will be able to reuse those objects.
Here, we could have a `CaseOption` or something like that, which would embed:
1. Adding case-file, import properties, and skip post processor options
2. Reading the network from the command line through a `read(line, context)` method (maybe platformConfig or importConfig as another parameter ?)
Plus maybe one more for footers?
Those 2 methods can be used by all tools which read a network from the command line.

What do you think ?

Reply:
I agree with your proposal. I implemented it as an interface which makes it easier to modify it in tests. What do you think?

Reply:
Nice! Is it not possible to go even further by having this same object add the options definitions to the `Options` object ? (skip-postproc, import parameters, and case-file)
I don't know, maybe it is not possible because the tools do not share exactly the same options.

Maybe we could have a more generic interface Option<T>, for example to read computation parameters from files in another evolution, but I am not sure having a common interface is really useful, so it could be done later if it becomes more apparently useful.

Reply:
This is a good idea but I think I will do it in another PR since this one is focused on conversionOption